### ISSUE TITLE
Support for multi-layers encoder/decoders

### ISSUE THREAD

**Issue by gidim** [EdinburghNLP/nematus#19]
Hi, 

Is there any plans to add support for multi-layers of GRUs?
It has proven to be effective for other seq2seq tasks.

Thanks

**Comment by rsennrich** [comment]
Hello,

yes, we are hoping to have multi-layer GRUs at some point in the future.

**Comment by bastings** [comment]
@rsennrich is someone already doing this? otherwise I can do it

**Comment by rsennrich** [comment]
we're currently experimenting with different deep architectures, and will push this soon.

**Comment by lixiangnlp** [comment]
@rsennrich The current implement of decoder is in fact a conditional GRU. Is it essential for multi-layer decoder.  In fact, the current decoder consists of two GRU layers and an attention layer. Do you have any experience about improving the performance using the multi-layer?

**Comment by rsennrich** [comment]
there are now several variants of deep architectures in the master branch. Experimental results will follow soon - I currently recommend adding depth via the parameters "--enc_recurrence_transition_depth" and "--dec_base_recurrence_transition_depth".

**Comment by rsennrich** [comment]
there's some more documentation on deep models at https://github.com/Avmb/deep-nmt-architectures

The full paper will be published at WMT, but as a preview, here's selected results (EN->DE WMT17 parallel training data; newstest2016 test set):

shallow-shallow: 29.3
deeptransition: 30.1
bideep-bideep-rGRU: 30.6
bideep-bideep-rGRU-large: 31.0
